### DISCUSSION TITLE
serve different layout by useragent with SSG

### DISCUSSION THREAD

**Discussion by pinkiebala** [vercel/next.js#14968] [Help]
I want to serve different layout by useragent (i.e: webview),
and wonder if there's a way to generate static page with `hybrid` prefix?, ie:

```
posts
> [id].html
> [id].js
> 1.html
> 1.html.hybrid
> 1.json
> 1.json.hybrid
```

The difference is the value passed with prop `hybrid`

``` tsx
export const getStaticProps: GetStaticProps<Props, Path> = async ({
  params
}) => {
  return { props: { 
    article: getArticle(params?.id),
    hybrid: true
 } };
};

const Post = ({article, hybrid}) => {
  if(hybrid || typeof window !== 'undefined' && navigator.userAgent.includes(APP_USERAGENT)){
    return(
      <HybridLayout>
        ...
        <Link href={{pathname: '/posts/[id]', query: {id: 2} }}>
      </HybridLayout>
    )
  }
  return (
    <Layout>
      ...
      <Link href={{pathname: '/posts/[id]', query: {id: 2} }}>
    </Layout>
  )
}
```

On the client-side, I'll use `navigator.userAgent` as the condition, so I expect even I `Link` to a non-hybrid page, React will hydrate as expected.

On the server-side, I'll use a custom server to check userAgent of request and serve different HTML.

**Comment by karandwivedi42** [comment]
Is this possible now with server components?
